std/machines/small_field/pointer_arith.asm

```diff
@@ -0,0 +1,35 @@
+/// Creates constraints that increments a pointer by one full 32-bit word.
+///
+/// The pointers are given in 2 16-bit limbs. This function return the polynomials constraining
+/// (int(post_high) * 0x10000 + int(post_low)) to (int(pre_high) * 0x10000 + int(pre_low) + 4).
+///
+/// The returned constraints are not automatically added to the constraints set. This is so the
+/// caller will be able to latch-disable them in rows they aren't needed.
+///
+/// This constr function introduces 2 new witness column and some helper constraints.
+///
+/// Preconditions: pre_low must be 16-bit aligned and in range [0, 0xfffc].
```

Comment:
This is quite important, maybe move it up a bit?

Reply:
Done.